### PR TITLE
Devoncarew scan cache

### PR DESCRIPTION
**Pull request by devoncarew**
@pq, @keertip, some initial support for scanning the pub cache. I'll follow up with some more functionality (and tests and docs) in later PRs. How does the API look, and do you see any glaring issues or holes w/ it?

### CODE REVIEW

File: lib/pubcache.dart [hunk 1]
```diff
@@ -4,4 +4,193 @@
...
+  }
+
+  List<PackageRef> getPackageRefs() {
+    if (_packageRefs == null) _parsePubspec();
+    return _packageRefs;
+  }
+
+  String toString() => '${name} ${version}';
+
+  void _parsePubspec() {
```
Comment (pq) [inline lib/pubcache.dart]: Maybe _parsePubspecLock()?

File: pubspec.yaml [hunk 2]
```diff
@@ -7,7 +7,10 @@ homepage: https://github.com/dart-lang/pubcache
 environment:
   sdk: '>=1.0.0 <2.0.0'
 
-#dependencies:
-#  lib_name: any
+dependencies:
+  path: any
+  pub_semver: any
+  yaml: any
+
```
Comment (pq) [inline pubspec.yaml:14]: Way to play nice with others! :)

**Comment by pq** [comment]
LGTM!

File: lib/pubcache.dart (hunk 1 above)
Comment (devoncarew) [inline lib/pubcache.dart]: done

File: pubspec.yaml (hunk 2 above)
Comment (devoncarew) [inline pubspec.yaml:14]: urk, this should probably be tightened up

**Comment by keertip** [comment]
Looks good!

**Pull request merged**